test/junit/scala/OptionJavaTest.java

```diff
@@ -0,0 +1,217 @@
+
+import scala.jdk.javaapi.CollectionConverters;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class OptionJavaTest {
+
+    private Option<Integer> someIntegerOpt = Option.apply(1);
+    private Option<Integer> noneIntegerOpt = Option.apply(null);
+
+    /*
+     * Compiler error: incompatible types: inference variable A has
+     * incompatible bounds:
+     *   equality constraints: scala.Int
+     *   lower bounds: java.lang.Integer
+     */
+    // private Option<Int> someScalaIntOpt = Option.apply(1);
+    private Option<Int> noneScalaIntOpt = Option.apply(null);
+
+    private Option<String> someStringOpt = Option.apply("hi");
+    private Option<String> noneStringOpt = Option.apply(null);
+
+    @Test
+    public void testSomeInteger() {
+        assertEquals(scala.Some.apply(1), someIntegerOpt);
+        assertTrue(someIntegerOpt.isDefined());
+        assertFalse(someIntegerOpt.isEmpty());
+        assertTrue(someIntegerOpt.nonEmpty());
+        assertEquals(1, someIntegerOpt.get().intValue());
+    }
+
+    @Test
+    public void testNoneInteger() {
+        assertEquals(scala.None$.MODULE$, noneIntegerOpt);
+        assertEquals(scala.Option$.MODULE$.empty(), noneIntegerOpt);
+        assertFalse(noneIntegerOpt.isDefined());
+        assertTrue(noneIntegerOpt.isEmpty());
+        assertFalse(noneIntegerOpt.nonEmpty());
+    }
+
+    @Test(expected = java.util.NoSuchElementException.class)
+    public void testGetNoneIntegerFailure() {
+        noneIntegerOpt.get();
+    }
+
+    @Test
+    public void testNoneScalaInt() {
+        assertEquals(scala.None$.MODULE$, noneScalaIntOpt);
+        assertEquals(scala.Option$.MODULE$.empty(), noneScalaIntOpt);
+        assertFalse(noneScalaIntOpt.isDefined());
+        assertTrue(noneScalaIntOpt.isEmpty());
+        assertFalse(noneScalaIntOpt.nonEmpty());
+    }
+
+    @Test(expected = java.util.NoSuchElementException.class)
+    public void testGetNoneScalaIntFailure() {
+        noneScalaIntOpt.get();
+    }
+
+    @Test
+    public void testSomeString() {
+        assertEquals(scala.Some.apply("hi"), someStringOpt);
+        assertTrue(someStringOpt.isDefined());
+        assertFalse(someStringOpt.isEmpty());
+        assertTrue(someStringOpt.nonEmpty());
+        assertEquals("hi", someStringOpt.get());
+    }
+
+    @Test
+    public void testNoneString() {
+        assertEquals(scala.None$.MODULE$, noneStringOpt);
+        assertEquals(scala.Option$.MODULE$.empty(), noneStringOpt);
+        assertFalse(noneStringOpt.isDefined());
+        assertTrue(noneStringOpt.isEmpty());
+        assertFalse(noneStringOpt.nonEmpty());
+    }
+
+    @Test(expected = java.util.NoSuchElementException.class)
+    public void testGetNoneStringFailure() {
+        noneStringOpt.get();
+    }
+
+    @Test
+    public void testMap() {
+        Option<Integer> nextIntegerOpt = Option.apply(2);
+
+        assertEquals(nextIntegerOpt, someIntegerOpt.map((Integer _i) -> 2));
```

Comment:
can javac infer the parameter type in the lambda?

Reply:
I believe so.  Should I add more tests without the annotation? Or do you want me to remove the annotation?

Reply:
I think having a regression test to ensure javac's inference keeps working is valuable, so i'd remove all the explicit parameter types if possible.

Reply:
Ok, will do.